crates/transport_tx5/src/lib.rs

```diff
@@ -0,0 +1,311 @@
+                        Ok(data.to_vec())
+                    })
+                }),
+                Arc::new(move |peer_url, data| {
+                    // process sent preflight data
+                    let peer_url = peer_url.clone();
+                    let pre_send = pre_send.clone();
+                    Box::pin(async move {
+                        let (s, r) = tokio::sync::oneshot::channel();
+                        // kitsune2 expects this to be sent in the normal
+                        // "recv_data" handler, so we need another task
+                        // to forward that.
+                        //
+                        // If the app is too slow processing incoming
+                        // preflights, reject it to close the connection.
+                        pre_send.try_send((peer_url, data, s)).map_err(
+                            |_| std::io::Error::other("app overloaded"),
+                        )?;
+                        r.await.map_err(|_| {
+                            std::io::Error::other("channel closed")
+                        })?
+                    })
+                }),
+            )),
+            backend_module: tx5::backend::BackendModule::LibDataChannel,
+            backend_module_config: Some(
+                tx5::backend::BackendModule::LibDataChannel.default_config(),
+            ),
+            ..Default::default()
+        });
+
+        let (ep, ep_recv) = tx5::Endpoint::new(tx5_config);
+        let ep = Arc::new(ep);
+
+        if let Some(local_url) = ep
+            .listen(
+                tx5::SigUrl::parse(&config.server_url).map_err(|e| {
+                    K2Error::other_src("parsing tx5 sig url", e)
+                })?,
+            )
+            .await
+        {
+            handler
+                .new_listening_address(Url::from_str(local_url.as_ref())?)
+                .await;
+        }
+
+        let pre_task = tokio::task::spawn(pre_task(handler.clone(), pre_recv));
+
+        let evt_task =
+            tokio::task::spawn(evt_task(handler, ep.clone(), ep_recv));
+
+        let out: DynTxImp = Arc::new(Self {
+            ep,
+            pre_task,
+            evt_task,
+        });
+
+        Ok(out)
+    }
+}
+
+impl TxImp for Tx5Transport {
+    fn url(&self) -> Option<Url> {
+        self.ep
+            .get_listening_addresses()
+            .first()
+            .and_then(|u| Url::from_str(u.as_ref()).ok())
+    }
+
+    fn disconnect(
+        &self,
+        peer: Url,
+        _payload: Option<(String, bytes::Bytes)>,
+    ) -> BoxFut<'_, ()> {
+        Box::pin(async move {
+            if let Ok(peer) = tx5::PeerUrl::parse(&peer) {
+                self.ep.close(&peer);
+            }
+        })
+    }
+
+    fn send(&self, peer: Url, data: bytes::Bytes) -> BoxFut<'_, K2Result<()>> {
+        Box::pin(async move {
+            let peer = peer.to_peer_url()?;
+            // this would be more efficient if we retool tx5 to use bytes
```

Comment:
Should this be a TODO or a task?

Reply:
This is part of the refactor tx5 option. We'd decided not to do it now. Do you think we should create an issue for it?

Reply:
I think it'd be good to track it even if we aren't going to do it right now